shell/platform/android/io/flutter/embedding/engine/dynamicfeatures/PlayStoreDynamicFeatureManager.java

```diff
@@ -0,0 +1,222 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.dynamicfeatures;
+
+import android.app.Application;
+import android.content.Context;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageManager;
+import android.content.pm.PackageManager.NameNotFoundException;
+import android.content.res.AssetManager;
+import android.security.NetworkSecurityPolicy;
+import android.util.Log;
+import android.widget.Toast;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.lifecycle.AndroidViewModel;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import com.google.android.play.core.splitinstall.SplitInstallException;
```

Comment:
We don't have to figure this out now, and you're probably already thinking about this since it implements an interface. But since this is all dead code for devs in China, we should figure out some sort of compile time or R8 time mechanism to let them strip all this out of their engine .so.

Reply:
Preliminary APK size comparisons point to a 30kb gain from play core + split aot code. I expect this number may be able to be reduced further as I realized I did not use a LTO dart for the comparison.

Reply:
Oops, the above comment probably belongs in the other thread about apk sizes.

Reply:
Just did another measurement, this time 28,935 bytes with a full LTO build. I'll take a look at source.